sopel/modules/reddit.py

```diff
@@ -206,3 +208,49 @@ def get_channel_sfw(bot, trigger):
         bot.say('%s is flagged as SFW' % channel)
     else:
         bot.say('%s is flagged as NSFW' % channel)
+
+
+@require_chanmsg('.setspoilfree is only permitted in channels')
+@commands('setspoilerfree', 'setspoilfree')
+@example('.setspoilfree true')
+@example('.setspoilfree false')
+def set_channel_spoiler_free(bot, trigger):
+    """
+    Sets the Spoiler-Free status (true or false) for the current channel.
+    Defaults to false.
+    """
+    if bot.channels[trigger.sender].privileges[trigger.nick] < OP:
+        return
+    else:
+        param = 'true'
+        if trigger.group(2) and trigger.group(3):
+            param = trigger.group(3).strip().lower()
+        spoiler_free = param == 'true'
+        bot.db.set_channel_value(trigger.sender, 'spoiler_free', spoiler_free)
+        if spoiler_free:
+            bot.reply('Got it. %s is now flagged as spoiler-free.' % trigger.sender)
+        else:
+            bot.reply('Got it. %s is now flagged as spoilers-allowed.' % trigger.sender)
+
+
+@commands('getspoilerfree', 'getspoilfree')
+@example('.getspoilfree [channel]')
+def get_channel_spoiler_free(bot, trigger):
```

Comment:
Is there a compelling reason this needs to be two separate commands? `.spoilerfree` could print the current state, and `.spoilerfree on|off` could set it.

Reply:
All I did was mirror the sfw setting commands. Setting only works in that channel; getting works from anywhere by specifying the channel name.